openedx/edx-django-utils
# Feat  add dd and otel celery tasks

**Description:**

Adds an initialize_celery_monitoring util to allow celery tasks to be monitored by the Open Telemetry and Datadog providers. edx-platform PR to follow.

**JIRA:**

https://github.com/edx/edx-arch-experiments/issues/584

**Testing instructions:**

1. Set this package to be installed locally, check out this PR
2. 3. Run your edx-platform off this branch: https://github.com/openedx/edx-platform/pull/34673 to use the implementation required from the package.
3. set your OPENEDX_TELEMETRY django setting to be [`edx_django_utils.monitoring.DataDogBackend`]
4. follow https://2u-internal.atlassian.net/wiki/spaces/~840928901/pages/813793298/OpenTelemetry+New+Relic+and+Datadog for devstack setup for DD, but set up the DD agent on CMS, not lms
5. Follow https://2u-internal.atlassian.net/wiki/spaces/ENGAGE/pages/395673715/How+to+configure+Celery+worker+on+LMS+CMS+Devstack to set up celery workers on cms.
4. Export a course
6. View the export on DD with a celery task

**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:**

List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.
